apps/emqx/src/emqx_frame.erl

```diff
@@ -117,6 +119,15 @@ initial_parse_state() ->
 initial_parse_state(Options) when is_map(Options) ->
     ?NONE(maps:merge(?DEFAULT_OPTIONS, Options)).
 
+%%--------------------------------------------------------------------
+%% Update Parse State
+%%--------------------------------------------------------------------
+
+update_parse_state(#{proto_ver := ProtoVer}, {ConnState, Opts}) ->
```

Comment:
maybe update should be internally and throw in the entire updates state instead of just a reason